src/plugins/data/common/index_patterns/fields/field_list.ts

```diff
@@ -44,87 +45,79 @@ export type CreateIndexPatternFieldList = (
 
-    specs.map((field) => this.add(field));
-  }
+    public readonly getAll = () => [...this.byName.values()];
+    public readonly getByName = (name: IndexPatternField['name']) => this.byName.get(name);
+    public readonly getByType = (type: IndexPatternField['type']) => [
+      ...(this.groups.get(type) || new Map()).values(),
+    ];
+    public readonly add = (field: FieldSpec) => {
+      const newField = new IndexPatternField(field, this.calcDisplayName(field.name));
+      this.push(newField);
+      this.setByName(newField);
+      this.setByGroup(newField);
+    };
 
-  public readonly getAll = () => [...this.byName.values()];
-  public readonly getByName = (name: IndexPatternField['name']) => this.byName.get(name);
-  public readonly getByType = (type: IndexPatternField['type']) => [
-    ...(this.groups.get(type) || new Map()).values(),
-  ];
-  public readonly add = (field: FieldSpec) => {
-    const newField = new IndexPatternField(
-      this.indexPattern,
-      field,
-      this.calcDisplayName(field.name),
-      this.onNotification
-    );
-    this.push(newField);
-    this.setByName(newField);
-    this.setByGroup(newField);
-  };
+    public readonly remove = (field: IFieldType) => {
+      this.removeByGroup(field);
+      this.byName.delete(field.name);
 
-  public readonly remove = (field: IFieldType) => {
-    this.removeByGroup(field);
-    this.byName.delete(field.name);
+      const fieldIndex = findIndex(this, { name: field.name });
+      this.splice(fieldIndex, 1);
+    };
 
-    const fieldIndex = findIndex(this, { name: field.name });
-    this.splice(fieldIndex, 1);
-  };
+    public readonly update = (field: FieldSpec) => {
+      const newField = new IndexPatternField(field, this.calcDisplayName(field.name));
+      const index = this.findIndex((f) => f.name === newField.name);
+      this.splice(index, 1, newField);
+      this.setByName(newField);
+      this.removeByGroup(newField);
+      this.setByGroup(newField);
+    };
 
-  public readonly update = (field: FieldSpec) => {
-    const newField = new IndexPatternField(
-      this.indexPattern,
-      field,
-      this.calcDisplayName(field.name),
-      this.onNotification
-    );
-    const index = this.findIndex((f) => f.name === newField.name);
-    this.splice(index, 1, newField);
-    this.setByName(newField);
-    this.removeByGroup(newField);
-    this.setByGroup(newField);
-  };
+    public readonly removeAll = () => {
+      this.length = 0;
+      this.byName.clear();
+      this.groups.clear();
+    };
 
-  public readonly removeAll = () => {
-    this.length = 0;
-    this.byName.clear();
-    this.groups.clear();
-  };
+    public readonly replaceAll = (spcs: FieldSpec[]) => {
+      this.removeAll();
+      spcs.forEach(this.add);
+    };
 
-  public readonly replaceAll = (specs: FieldSpec[]) => {
-    this.removeAll();
-    specs.forEach(this.add);
-  };
+    public toSpec({
+      getFormatterForField,
```

Comment:
then we don't need this one here, as we already have the serialized field format on the field.